api/v3/Activity.php

```diff
@@ -471,7 +471,7 @@ function _civicrm_api3_activity_get_formatResult($params, $activities, $options)
         $dao = CRM_Core_DAO::executeQuery("SELECT activity_id, case_id FROM civicrm_case_activity WHERE activity_id IN (%1)",
           array(1 => array(implode(',', array_keys($activities)), 'String', CRM_Core_DAO::QUERY_FORMAT_NO_QUOTES)));
         while ($dao->fetch()) {
-          $activities[$dao->activity_id]['case_id'] = $dao->case_id;
+          $activities[$dao->activity_id]['case_id'][] = $dao->case_id;
```

Comment:
OK, so repeating aloud to check my comprehension:
 * Activities are *normally* associated with zero or one cases, but in the scenario of linked-cases, you can have an activity which lives in multiple cases.
 * The old/existing API contract returns an `int` or null, which is fine for the normal scenario. The new/proposed API contract returns an `array` of `int`s, which is more generally correct.
 * __Problem__: Changing the data-structure of `case_id` will break any API consumers who use it.
 * __Suggestion__: The `Activity` API should support two different return notations, `case_id` (singular, legacy behavior) or `case_ids` (array, new behavior). 
 * __Suggestion__: Whether returning `case_id` or `case_ids`, the query which reads `civicrm_case_activity` should include an `ORDER BY` clause to ensure that the result is deterministic.